### PR TITLE
first cut at error report ui

### PR DESCRIPTION
**Pull request by mwhudson**
What this conspicuously lacks is any support for submitting reports to Daisy or reporting bugs via apport. We'll decide what to do about those later I guess. You can play around with various kinds of failures in dry-run mode by using control-e, control-r, control-u or setting SUBIQUITY_DEBUG.

### CODE REVIEW

**Comment by mwhudson** [comment]
> Need to check more things though still.

Thanks for the comments so far!

File: subiquity/controllers/error.py [hunk 1]
```diff
@@ -132,11 +177,48 @@ def set_meta(self, key, value):
...
+        return self.meta.get("seen", False)
+
+    @property
+    def persistent_details(self):
+        """Return fs-label, path-on-fs to report."""
+        # Not sure if this is more or less sane than shelling out to
+        # findmnt(1).
+        looking_for = os.path.abspath(
+            os.path.normpath(self.controller.crash_directory))
+        for line in open('/proc/self/mountinfo'):
```
[the remark this answers is not in the document]
Comment (mwhudson) [reply subiquity/controllers/error.py]: I guess we can read the mountinfo in one go. I think we'd have to be severely unlucky to hit this race but why take the risk.

File: subiquity/controllers/error.py [hunk 2]
```diff
@@ -104,17 +126,40 @@ def added_info(fut):
             try:
                 fut.result()
             except Exception:
-                self.state = ErrorReportState.ERROR
+                self.state = ErrorReportState.ERROR_GENERATING
```
[the remark this answers is not in the document]
Comment (mwhudson) [reply subiquity/controllers/error.py]: Everything that is in the discover log is also in the "regular" log and so available there. We could attach the disk storage we get back from curtin separately, not sure how to do the "unmerged storage config" that curtin logs though other than some very specific log handling (probably better to make a curtin change to make it accessible I guess).

File: subiquity/core.py [hunk 3]
```diff
@@ -130,6 +139,13 @@ def unhandled_input(self, key):
         if key == 'f1':
             if not self.ui.right_icon.showing_something:
                 self.ui.right_icon.open_pop_up()
+        elif self.opts.dry_run and key in ['ctrl e', 'ctrl r']:
+            interrupt = key == 'ctrl e'
+            try:
+                1/0
+            except ZeroDivisionError:
+                self.make_apport_report(
+                    ErrorReportKind.UNKNOWN, "example", interrupt=interrupt)
```
[the remark this answers is not in the document]
Comment (mwhudson) [reply subiquity/core.py]: C-u just executes 1/0. This is intended to mimic an internal error that subiquity catches, more in the style of a block probe failure. Overriding excepthook might be better than try:/except: around loop.run as we currently do, but not very much better I'd wager.

File: subiquity/controllers/error.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cyphermox) [reply subiquity/controllers/error.py]: +1 for using readlines().

File: subiquity/controllers/error.py [hunk 4]
```diff
@@ -57,7 +64,7 @@ class ErrorReport:
 
     @classmethod
     def new(cls, controller, kind):
-        base = "installer.{:.9f}.{}".format(time.time(), kind.name.lower())
+        base = "{:.9f}.{}".format(time.time(), kind.name.lower())
```
[the remark this answers is not in the document]
Comment (mwhudson) [reply subiquity/controllers/error.py]: I thought I replied here, but my thinking was to have the crash directory sort usefully. I'm not super happy with the current names, they are a bit long, but well. Easy to change later.

**Comment by mwhudson** [comment]
Merging after the review from @cyphermox but still happy to hear any more thoughts from @xnox

**Pull request merged**